### PR TITLE
support for the UMD package format

### PR DESCRIPTION
**Pull request by emmanueltouzery**
I saw in https://github.com/bvaughn/react-window/issues/232 that you'd consider a PR to support UMD package generation, and it would be very important for us to have a UMD package of react-window honestly.

So here's a PR. I tested the generated UMD with our app and it seems to work just fine. There are two open questions maybe:

1. I tested by copying the generated JS files in our app. We just need to be sure that the files will indeed be in the NPM package. I believe they'll be, but...

2. If you want to just provide UMD support with the minimal extra dependencies, maybe you don't want to "uglify" (minimize) the output. For our purpose we don't need a minimized UMD, but some people may expect it. So I made two commits, the uglify support is in the second commit. So you can take it in, or not. I checked and formik for instance ships minified UMDs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -61,7 +63,8 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.0.0",
-    "memoize-one": ">=3.1.1 <6"
+    "memoize-one": ">=3.1.1 <6",
+    "rollup-plugin-replace": "^2.2.0"
```
Comment (bvaughn) [inline package.json]: This should be a dev dependency, not a dependency

File: rollup.config.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import path from 'path';
 import babel from 'rollup-plugin-babel';
+import replace from 'rollup-plugin-replace';
 import commonjs from 'rollup-plugin-commonjs';
 import nodeResolve from 'rollup-plugin-node-resolve';
+import { uglify } from 'rollup-plugin-uglify';
```
Comment (TrySound) [inline rollup.config.js]: Use terser please

File: rollup.config.js [hunk 3]
```diff
@@ -47,4 +49,58 @@ export default [
...
+    plugins: [
+      babel({
+        runtimeHelpers: true,
+        plugins: [['@babel/transform-runtime', { useESModules: true }]],
+        sourceMaps: true,
+      }),
+      nodeResolve(),
+      commonjs(),
+      replace({
+        "process.env.NODE_ENV": "'development'"
```
Comment (TrySound) [inline rollup.config.js]: JSON.stringify('development')

File: rollup.config.js [hunk 4]
```diff
@@ -47,4 +49,58 @@ export default [
...
+      sourcemap: true,
+      name: 'ReactWindow',
+      globals: {
+        'react': 'React'
+      }
+    },
+    external: ['react'],
+    plugins: [
+      babel({
+        runtimeHelpers: true,
```
Comment (TrySound) [inline rollup.config.js]: Reuse esm babel config

File: rollup.config.js [hunk 5]
```diff
@@ -47,4 +49,58 @@ export default [
...
+
+  {
+    input,
+    output: {
+      file: pkg.umddev,
+      format: 'umd',
+      sourcemap: true,
+      name: 'ReactWindow',
+      globals: {
+        'react': 'React'
```
Comment (TrySound) [inline rollup.config.js]: Define globals separately and reuse

File: rollup.config.js [hunk 6]
```diff
@@ -47,4 +49,58 @@ export default [
...
+      }
+    },
+    external: ['react'],
```
Comment (TrySound) [inline rollup.config.js]: Object.keys(globals)

File: package.json [hunk 7]
```diff
@@ -37,6 +37,8 @@
   "main": "dist/index.cjs.js",
   "module": "dist/index.esm.js",
   "files": ["dist", "src/*.js"],
+  "umddev": "dist/index-dev.umd.js",
+  "umdprod": "dist/index-prod.umd.js",
```
Comment (TrySound) [inline package.json]: Not sure about this. Maybe hardcode in config?

**Comment by emmanueltouzery** [comment]
ok, all clear! I'll resubmit with your suggestions + the yarn lint failures.

**Comment by TrySound** [comment]
Just push in this branch

**Comment by emmanueltouzery** [comment]
I pushed a new version!

**Comment by bvaughn** [comment]
Thanks for the code review, @TrySound.

Published 1.8.5

**Pull request merged**